### ISSUE TITLE
Calling `get_palette` before `setup` seems to raise an error

### ISSUE THREAD

**Issue by rawhat** [catppuccin/nvim#535]
### Description

It seems like the recent change to set `default_options` instead of `options` causes this to error.  It raises here https://github.com/catppuccin/nvim/blob/main/lua/catppuccin/palettes/init.lua#L7 since `O` is `nil`.

I was able to fix this by changing that line to `default_options`, but I'm not sure if that's the correct approach. That property does get set after setup.

For reference, I pull in the palettes to pass to some custom highlights in setup:

```lua
local colors = require("catppuccin.palettes").get_palette(palette or "mocha")
require("catppuccin").setup({
  flavour = palette or "mocha",
  custom_highlights = {
    cjsxElement = { link = "@constructor" },
    cjsxAttribProperty = { link = "@tag.attribute" },
    MatchParenCur = { fg = colors.yellow, style = { "bold" } },
    MatchParen = { bg = colors.base, fg = colors.yellow, style = { "bold" } },
    LspInlayHint = { link = "Comment" },
  },
  ...
```

### Neovim version

```markdown
NVIM v0.10.0-dev-546+g3bf887f6e0
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
```

### Terminal and multiplexer

wezterm 20230622-212623-9a24f61c with tmux 3.3a

### Catppuccin version / branch / rev

main

### Steps to reproduce

The snippet is listed above.

### Expected behavior

I'd expect it to not throw an error.

### Actual behavior

It does.

**Comment by nullchilly** [comment]
Can you try again?

**Comment by canvanaut** [comment]
Yeah I've encountered the same issue. I've refactored my custom_highlights into a function, but am unsure how to set the value for an integration, e.g. `integrations.navic.custom_bg = colours.base`.

**Comment by nullchilly** [comment]
Sorry for the inconveniece, maybe fixed in https://github.com/catppuccin/nvim/commit/841d8abf3be39de833d95a592a1fbbb1b9851296

For a better way, you can use the function

```lua
require("catppuccin").setup({
  flavour = palette or "mocha",
  custom_highlights = function(colors)
    return {
      cjsxElement = { link = "@constructor" },
      cjsxAttribProperty = { link = "@tag.attribute" },
      MatchParenCur = { fg = colors.yellow, style = { "bold" } },
      MatchParen = { bg = colors.base, fg = colors.yellow, style = { "bold" } },
      LspInlayHint = { link = "Comment" },
    }
  end
}
```

About the `navic` integration, yeah I'm troubled by it too. Maybe we should accept integration function...

**Comment by nullchilly** [comment]
> > Can you try again?
> 
> This works great for me! Can hold off on closing if the above question is still relevant. Thank you 😄

Will add this to backwards compatibility check asap, thank you so much!!

**Comment by canvanaut** [comment]
Sorry for hijacking this thread, but yeah I've just resorted to hard coding the value for now. I've seen some other themes dynamically convert strings but not sure if this is the best way either.
